lib/schemes/nested.js

```diff
@@ -12,6 +12,48 @@ module.exports = {
                 entity.modName ? modFolder : '');
 
         return path.join(folder,
-            naming.stringify(entity) + '.' + tech);
+            naming.stringify(entity) + (tech ? '.' + tech : ''));
+    },
+    parse: function(str, options) {
+        options || (options = {});
+        var naming = bemNaming(options.naming),
+            splittedPath = str.split(path.sep).filter(Boolean),
+            file = splittedPath.pop(),
+            splittedFile = file.split('.'),
+            entity = naming.parse(splittedFile[0]),
+            tech = splittedFile.slice(1).join('.');
+
+        // support for paths without filename, e.g. 'b1/__e1/'
+        if (!entity) {
+            entity = naming.parse(splittedPath.join('') + splittedFile[0]);
+        }
+
+        if (splittedPath.length && entity.block !== splittedPath[0]) {
+            return;
+        }
+
+        // check if path is valid
+        for (var i = 1; i < splittedPath.length; i++) {
+            var chunk = splittedPath[i];
+
+            // __e1 or _m1
+            if (i === 1 &&
+                (chunk.indexOf(naming.elemDelim) !== 0 &&
+                chunk.indexOf(naming.modDelim) !== 0)
+            ) { return; }
+
+            // __e1/_m1
+            if (i === 2 &&
+                (splittedPath[i - 1].indexOf(naming.elemDelim) !== 0 ||
+                chunk.indexOf(naming.modDelim) !== 0)
+            ) { return; }
+
+            if (i > 2) { return; }
+        }
+
+        return {
+            entity: entity,
+            tech: tech
+        };
```

Comment:
yes, it's mine

yep, i can, but i afraid it useless and what would be changed if i do?